phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PhoenixMasterObserver.java

```diff
@@ -0,0 +1,187 @@
+            + CDCUtil.CdcStreamStatus.ENABLED.getSerializedValue() + "'";
+
+    // tableName, streamName, partitionId, parentId, startTime, endTime, startKey, endKey
+    private static final String PARTITION_UPSERT_SQL
+            = "UPSERT INTO " + SYSTEM_CDC_STREAM_NAME + "VALUES (?,?,?,?,?,?,?,?)";
+
+    private static final String PARENT_PARTITION_QUERY
+            = "SELECT PARTITION_ID FROM " + SYSTEM_CDC_STREAM_NAME +
+            " WHERE TABLE_NAME = ? AND STREAM_NAME = ? ";
+
+    private static final String PARENT_PARTITION_UPDATE_END_TIME_SQL
+            = "UPSERT INTO" + SYSTEM_CDC_STREAM_NAME + " (TABLE_NAME, STREAM_NAME, PARTITION_ID, " +
+            "PARTITION_END_TIME) VALUES (?,?,?,?)";
+
+    @Override
+    public Optional<MasterObserver> getMasterObserver() {
+        return Optional.of(this);
+    }
+
+    /**
+     * Update parent -> daughter relationship for CDC Streams.
+     * - find parent partition id using start/end keys of daughters
+     * - upsert partition metadata for the 2 daughters
+     * - update the end time on the parent's partition metadata
+     * @param c           the environment to interact with the framework and master
+     * @param regionInfoA the left daughter region
+     * @param regionInfoB the right daughter region
+     */
+    @Override
+    public void postCompletedSplitRegionAction(final ObserverContext<MasterCoprocessorEnvironment> c,
+                                               final RegionInfo regionInfoA, final RegionInfo regionInfoB) {
+        Configuration conf = c.getEnvironment().getConfiguration();
+        try {
+            Connection conn  = QueryUtil.getConnectionOnServer(conf);
+            // CDC will be enabled on Phoenix tables only
+            PTable phoenixTable = getPhoenixTable(conn, regionInfoA.getTable());
+            if (phoenixTable == null) {
+                LOGGER.info("Not a Phoenix Table, skipping partition metadata update.");
+                return;
+            }
+            // find streamName with ENABLED status
+            String tableName = phoenixTable.getName().getString();
+            PreparedStatement pstmt = conn.prepareStatement(STREAM_STATUS_QUERY);
+            pstmt.setString(1, tableName);
+            ResultSet rs = pstmt.executeQuery();
+            if (rs.next()) {
+                String streamName = rs.getString(1);
+                String parentPartitionID = getParentPartitionId(conn, tableName, streamName, regionInfoA, regionInfoB);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoA);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoB);
+                updateParentPartitionEndTime(conn, tableName, streamName, parentPartitionID, regionInfoA.getRegionId());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Unable to update CDC Stream Partition metadata: " + e);
```

Comment:
Additional log message to include regionA and regionB would be great for debugging

Reply:
done